caikit/core/module.py

```diff
@@ -1248,6 +1260,30 @@ def decorator(cls_):
                 cls_.MODULE_CLASS = classname
                 cls_.PRODUCER_ID = dm.ProducerId(cls_.MODULE_NAME, cls_.MODULE_VERSION)
 
+                # Tasks: check to see if a super-class has one as well and that they match:
+                tasks = {
+                    class_.TASK_CLASS
+                    for class_ in cls_.mro()
+                    if hasattr(class_, "TASK_CLASS")
+                }
+                if len(tasks) > 1:
+                    raise TypeError(
+                        f"Class {cls_} has multiple task definitions in class hierarchy"
+                    )
+                if tasks:
+                    parent_task = tasks.pop()
+                    if task and task != parent_task:
+                        raise TypeError(
+                            f"Class {cls_} has task {task} but superclass has task {parent_task}"
+                        )
+                    cls_.TASK_CLASS = parent_task
+                else:
+                    if not task:
```

Comment:
Maybe stick a `warning` (or at least a `debug`) log statement here so that authors whose blocks don't have a `task` can easily identify them

Reply:
🌶️